### ISSUE TITLE
s3 support no longer working

### ISSUE THREAD

**Issue by alpapan** [GMOD/Apollo#2158]
When I updated to the git master, I noticed that the S3 support that previously worked, no longer does:

For example the trackList.json and seq/refSeqs.json that worked in the latest docker container, no longer works with git master because it appends the directory to the S3 url definition.

```
{
   "formatVersion" : 1,
   "names" : {
      "type" : "Hash",
      "url" : "https://s3.hortgenomics.science/jbrowse/almond/data/names/"
   },
   "refSeqs" : "https://s3.hortgenomics.science/jbrowse/almond/data/seq/sequence.fasta.fai",
   "tracks" : [
      {
         "category" : "Reference sequence",
         "faiUrlTemplate" : "https://s3.hortgenomics.science/jbrowse/almond/data/seq/sequence.fasta.fai",
         "key" : "Reference sequence",
         "label" : "DNA",
         "seqType" : "dna",
         "storeClass" : "JBrowse/Store/SeqFeature/IndexedFasta",
         "type" : "SequenceTrack",
         "urlTemplate" : "https://s3.hortgenomics.science/jbrowse/almond/data/seq/sequence.fasta",
         "useAsRefSeqStore" : 1
      },
      {
         "compress" : 0,
         "key" : "Official annotation",
         "label" : "Official annotation",
         "storeClass" : "JBrowse/Store/SeqFeature/NCList",
         "style" : {
            "className" : "feature"
         },
         "trackType" : null,
         "type" : "FeatureTrack",
         "urlTemplate" : "https://s3.hortgenomics.science/jbrowse/almond/data/tracks/Official annotations/{refseq}/trackData.json"
      },
      {
         "compress" : 0,
         "key" : "Repeats",
         "label" : "Repeats",
         "storeClass" : "JBrowse/Store/SeqFeature/NCList",
         "style" : {
            "className" : "feature"
         },
         "trackType" : null,
         "type" : "FeatureTrack",
         "urlTemplate" : "https://s3.hortgenomics.science/jbrowse/almond/data/tracks/Repeats/{refseq}/trackData.json"
      }
   ]
}
```

Cannot access first() element from an empty List. Stacktrace follows:
java.util.NoSuchElementException: Cannot access first() element from an empty List
        at org.bbop.apollo.PreferenceService.$tt__setCurrentOrganism(PreferenceService.groovy:199)
        at org.bbop.apollo.AnnotatorController.$tt__setCurrentOrganism(AnnotatorController.groovy:610)
        at grails.plugin.cache.web.filter.PageFragmentCachingFilter.doFilter(PageFragmentCachingFilter.java:198)
        at grails.plugin.cache.web.filter.AbstractFilter.doFilter(AbstractFilter.java:63)
        at org.apache.shiro.web.servlet.AbstractShiroFilter.executeChain(AbstractShiroFilter.java:449)
        at org.apache.shiro.web.servlet.AbstractShiroFilter$1.call(AbstractShiroFilter.java:365)
        at org.apache.shiro.subject.support.SubjectCallable.doCall(SubjectCallable.java:90)
        at org.apache.shiro.subject.support.SubjectCallable.call(SubjectCallable.java:83)
        at org.apache.shiro.subject.support.DelegatingSubject.execute(DelegatingSubject.java:383)
        at org.apache.shiro.web.servlet.AbstractShiroFilter.doFilterInternal(AbstractShiroFilter.java:362)
        at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:125)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
2019-05-18 16:32:31,140 [http-nio-8080-exec-5] ERROR errors.GrailsExceptionResolver  - NoSuchElementException occurred when processing request: [GET] /apollo/annotator/setCurrentOrganism/1184
Cannot access first() element from an empty List. Stacktrace follows:
java.util.NoSuchElementException: Cannot access first() element from an empty List
        at org.bbop.apollo.PreferenceService.$tt__setCurrentOrganism(PreferenceService.groovy:199)
        at org.bbop.apollo.AnnotatorController.$tt__setCurrentOrganism(AnnotatorController.groovy:610)
        at grails.plugin.cache.web.filter.PageFragmentCachingFilter.doFilter(PageFragmentCachingFilter.java:198)
        at grails.plugin.cache.web.filter.AbstractFilter.doFilter(AbstractFilter.java:63)
        at org.apache.shiro.web.servlet.AbstractShiroFilter.executeChain(AbstractShiroFilter.java:449)
        at org.apache.shiro.web.servlet.AbstractShiroFilter$1.call(AbstractShiroFilter.java:365)
        at org.apache.shiro.subject.support.SubjectCallable.doCall(SubjectCallable.java:90)
        at org.apache.shiro.subject.support.SubjectCallable.call(SubjectCallable.java:83)
        at org.apache.shiro.subject.support.DelegatingSubject.execute(DelegatingSubject.java:383)
        at org.apache.shiro.web.servlet.AbstractShiroFilter.doFilterInternal(AbstractShiroFilter.java:362)
        at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:125)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
2019-05-18 16:33:24,127 [http-nio-8080-exec-2] INFO  apollo.UserController  - updateTrackListPreference
2019-05-18 16:33:24,129 [http-nio-8080-exec-2] INFO  apollo.UserController  - Added userOrganismPreference false
2019-05-18 16:33:28,649 [http-nio-8080-exec-1] WARN  apollo.TrackService  - File /scratch_cory/tmp/23-almond does not exist
2019-05-18 16:33:28,649 [http-nio-8080-exec-1] WARN  apollo.TrackService  - File /scratch_cory/tmp/23-almond can not be read
2019-05-18 16:33:28,649 [http-nio-8080-exec-1] WARN  apollo.TrackService  - File /scratch_cory/tmp/23-almond can not be written to
2019-05-18 16:33:28,649 [http-nio-8080-exec-1] WARN  apollo.TrackService  - File /scratch_cory/tmp/23-almond/extendedTrackList.json does not exist
2019-05-18 16:33:28,649 [http-nio-8080-exec-1] WARN  apollo.TrackService  - File /scratch_cory/tmp/23-almond/extendedTrackList.json can not be read
2019-05-18 16:33:28,649 [http-nio-8080-exec-1] WARN  apollo.TrackService  - File /scratch_cory/tmp/23-almond/extendedTrackList.json can not be written to
2019-05-18 16:33:33,541 [http-nio-8080-exec-9] INFO  apollo.UserController  - updateTrackListPreference
2019-05-18 16:33:33,544 [http-nio-8080-exec-9] INFO  apollo.UserController  - Added userOrganismPreference false
2019-05-18 16:33:35,382 [http-nio-8080-exec-3] WARN  apollo.TrackService  - File /scratch_cory/tmp/23-almond does not exist
2019-05-18 16:33:35,382 [http-nio-8080-exec-3] WARN  apollo.TrackService  - File /scratch_cory/tmp/23-almond can not be read
2019-05-18 16:33:35,382 [http-nio-8080-exec-3] WARN  apollo.TrackService  - File /scratch_cory/tmp/23-almond can not be written to
2019-05-18 16:33:35,382 [http-nio-8080-exec-3] WARN  apollo.TrackService  - File /scratch_cory/tmp/23-almond/extendedTrackList.json does not exist
2019-05-18 16:33:35,383 [http-nio-8080-exec-3] WARN  apollo.TrackService  - File /scratch_cory/tmp/23-almond/extendedTrackList.json can not be read
2019-05-18 16:33:35,383 [http-nio-8080-exec-3] WARN  apollo.TrackService  - File /scratch_cory/tmp/23-almond/extendedTrackList.json can not be written to
java.io.FileNotFoundException: Genome fasta /scratch_cory/sysadmin/apollo/data/almond_s3/https:/s3.hortgenomics.science/jbrowse/almond/data/seq/sequence.fasta does not exist!
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:80)
        at org.codehaus.groovy.reflection.CachedConstructor.doConstructorInvoke(CachedConstructor.java:74)
        at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrap.callConstructor(ConstructorSite.java:84)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:247)
        at org.bbop.apollo.SequenceService.$tt__loadGenomeFasta(SequenceService.groovy:354)
        at org.bbop.apollo.SequenceService$_loadGenomeFasta_closure13.doCall(SequenceService.groovy)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1210)
        at groovy.lang.ExpandoMetaClass.invokeMethod(ExpandoMetaClass.java:1123)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1019)
        at groovy.lang.Closure.call(Closure.java:426)
        at groovy.lang.Closure.call(Closure.java:442)
        at org.codehaus.groovy.grails.orm.support.GrailsTransactionTemplate$2.doInTransaction(GrailsTransactionTemplate.groovy:90)
        at org.springframework.transaction.support.TransactionTemplate.execute(TransactionTemplate.java:133)
        at org.codehaus.groovy.grails.orm.support.GrailsTransactionTemplate.execute(GrailsTransactionTemplate.groovy:87)
        at org.bbop.apollo.SequenceService.loadGenomeFasta(SequenceService.groovy)
        at org.bbop.apollo.SequenceService$loadGenomeFasta$7.callCurrent(Unknown Source)
        at org.bbop.apollo.SequenceService.$tt__loadRefSeqs(SequenceService.groovy:277)
        at org.bbop.apollo.SequenceService$_loadRefSeqs_closure11.doCall(SequenceService.groovy)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1210)
        at groovy.lang.ExpandoMetaClass.invokeMethod(ExpandoMetaClass.java:1123)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1019)
        at groovy.lang.Closure.call(Closure.java:426)
        at groovy.lang.Closure.call(Closure.java:442)
        at org.codehaus.groovy.grails.orm.support.GrailsTransactionTemplate$2.doInTransaction(GrailsTransactionTemplate.groovy:90)
        at org.springframework.transaction.support.TransactionTemplate.execute(TransactionTemplate.java:133)
        at org.codehaus.groovy.grails.orm.support.GrailsTransactionTemplate.execute(GrailsTransactionTemplate.groovy:87)
        at org.bbop.apollo.SequenceService.loadRefSeqs(SequenceService.groovy)
        at org.bbop.apollo.SequenceService$loadRefSeqs$5.call(Unknown Source)
        at org.bbop.apollo.OrganismController.$tt__addOrganism(OrganismController.groovy:1096)
        at org.bbop.apollo.OrganismController$_addOrganism_closure8.doCall(OrganismController.groovy)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1210)
        at groovy.lang.ExpandoMetaClass.invokeMethod(ExpandoMetaClass.java:1123)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1019)
        at groovy.lang.Closure.call(Closure.java:426)
        at groovy.lang.Closure.call(Closure.java:442)
        at org.codehaus.groovy.grails.orm.support.GrailsTransactionTemplate$2.doInTransaction(GrailsTransactionTemplate.groovy:90)
        at org.springframework.transaction.support.TransactionTemplate.execute(TransactionTemplate.java:133)
        at org.codehaus.groovy.grails.orm.support.GrailsTransactionTemplate.execute(GrailsTransactionTemplate.groovy:87)
        at org.bbop.apollo.OrganismController.addOrganism(OrganismController.groovy)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.grails.web.servlet.mvc.MixedGrailsControllerHelper.invoke(MixedGrailsControllerHelper.java:154)
        at org.codehaus.groovy.grails.web.servlet.mvc.AbstractGrailsControllerHelper.handleAction(AbstractGrailsControllerHelper.java:375)
        at org.codehaus.groovy.grails.web.servlet.mvc.AbstractGrailsControllerHelper.executeAction(AbstractGrailsControllerHelper.java:252)
        at org.codehaus.groovy.grails.web.servlet.mvc.AbstractGrailsControllerHelper.handleURI(AbstractGrailsControllerHelper.java:205)
        at org.codehaus.groovy.grails.web.servlet.mvc.AbstractGrailsControllerHelper.handleURI(AbstractGrailsControllerHelper.java:126)
        at org.codehaus.groovy.grails.web.servlet.mvc.SimpleGrailsController.handleRequest(SimpleGrailsController.java:72)
        at org.springframework.web.servlet.mvc.SimpleControllerHandlerAdapter.handle(SimpleControllerHandlerAdapter.java:50)
        at org.codehaus.groovy.grails.web.servlet.GrailsDispatcherServlet.doDispatch(GrailsDispatcherServlet.java:347)
        at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:893)
        at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:967)
        at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:869)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:648)
        at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:843)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:292)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
        at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
        at grails.plugin.cache.web.filter.PageFragmentCachingFilter.doFilter(PageFragmentCachingFilter.java:198)
        at grails.plugin.cache.web.filter.AbstractFilter.doFilter(AbstractFilter.java:63)
        at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:344)
        at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:261)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:101)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:101)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:101)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
        at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:720)
        at org.apache.catalina.core.ApplicationDispatcher.processRequest(ApplicationDispatcher.java:466)
        at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:391)
        at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:318)
        at org.codehaus.groovy.grails.web.mapping.UrlMappingUtils.forwardRequestForUrlMappingInfo(UrlMappingUtils.java:178)
        at org.codehaus.groovy.grails.web.mapping.UrlMappingUtils.forwardRequestForUrlMappingInfo(UrlMappingUtils.java:144)
        at org.codehaus.groovy.grails.web.mapping.UrlMappingUtils.forwardRequestForUrlMappingInfo(UrlMappingUtils.java:135)
        at org.codehaus.groovy.grails.web.mapping.filter.UrlMappingsFilter.doFilterInternal(UrlMappingsFilter.java:216)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
        at org.codehaus.groovy.grails.web.servlet.mvc.GrailsWebRequestFilter.doFilterInternal(GrailsWebRequestFilter.java:69)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
        at org.codehaus.groovy.grails.web.filters.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:67)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
        at org.apache.shiro.web.servlet.AbstractShiroFilter.executeChain(AbstractShiroFilter.java:449)
        at org.apache.shiro.web.servlet.AbstractShiroFilter$1.call(AbstractShiroFilter.java:365)
        at org.apache.shiro.subject.support.SubjectCallable.doCall(SubjectCallable.java:90)
        at org.apache.shiro.subject.support.SubjectCallable.call(SubjectCallable.java:83)
        at org.apache.shiro.subject.support.DelegatingSubject.execute(DelegatingSubject.java:383)
        at org.apache.shiro.web.servlet.AbstractShiroFilter.doFilterInternal(AbstractShiroFilter.java:362)
        at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:125)
        at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:344)
        at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:261)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
        at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:85)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
        at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:344)
        at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:261)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:213)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:106)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:494)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:141)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79)
        at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:616)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:522)
        at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1095)
        at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:672)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1520)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1476)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
        at java.lang.Thread.run(Thread.java:748)
2019-05-18 16:34:06,479 [http-nio-8080-exec-3] ERROR apollo.OrganismController  - problem saving organism: java.io.FileNotFoundException: Genome fasta /scratch_cory/sysadmin/apollo/data/almond_s3/https:/s3.hortgenomics.science/jbrowse/almond/data/seq/sequence.fasta does not exist!
2019-05-18 16:34:07,950 [http-nio-8080-exec-5] INFO  apollo.UserController  - updateTrackListPreference
2019-05-18 16:34:07,955 [http-nio-8080-exec-5] INFO  apollo.UserController  - Added userOrganismPreference false
{Heap before GC invocations=7 (full 2):
 par new generation   total 1993600K, used 1993599K [0x00000004c0000000, 0x0000000547330000, 0x0000000547330000)
  eden space 1772096K, 100% used [0x00000004c0000000, 0x000000052c290000, 0x000000052c290000)
  from space 221504K,  99% used [0x0000000539ae0000, 0x000000054732fff8, 0x0000000547330000)
  to   space 221504K,   0% used [0x000000052c290000, 0x000000052c290000, 0x0000000539ae0000)
 concurrent mark-sweep generation total 6173504K, used 295215K [0x0000000547330000, 0x00000006c0000000, 0x00000007c0000000)
 Metaspace       used 126587K, capacity 134839K, committed 135168K, reserved 1167360K
  class space    used 14394K, capacity 16772K, committed 16872K, reserved 1048576K
1646.106: [GC (Allocation Failure) 1646.106: [ParNew: 1993599K->221504K(1993600K), 0.3990983 secs] 2288815K->638806K(8167104K), 0.3994672 secs] [Times: user=1.34 sys=0.37, real=0.40 secs]
Heap after GC invocations=8 (full 2):
 par new generation   total 1993600K, used 221504K [0x00000004c0000000, 0x0000000547330000, 0x0000000547330000)
  eden space 1772096K,   0% used [0x00000004c0000000, 0x00000004c0000000, 0x000000052c290000)
  from space 221504K, 100% used [0x000000052c290000, 0x0000000539ae0000, 0x0000000539ae0000)
  to   space 221504K,   0% used [0x0000000539ae0000, 0x0000000539ae0000, 0x0000000547330000)
 concurrent mark-sweep generation total 6173504K, used 417302K [0x0000000547330000, 0x00000006c0000000, 0x00000007c0000000)
 Metaspace       used 126587K, capacity 134839K, committed 135168K, reserved 1167360K
  class space    used 14394K, capacity 16772K, committed 16872K, reserved 1048576K
}
2019-05-18 16:34:09,974 [http-nio-8080-exec-4] WARN  apollo.TrackService  - File /scratch_cory/tmp/23-almond does not exist
2019-05-18 16:34:09,974 [http-nio-8080-exec-4] WARN  apollo.TrackService  - File /scratch_cory/tmp/23-almond can not be read
2019-05-18 16:34:09,974 [http-nio-8080-exec-4] WARN  apollo.TrackService  - File /scratch_cory/tmp/23-almond can not be written to
2019-05-18 16:34:09,974 [http-nio-8080-exec-4] WARN  apollo.TrackService  - File /scratch_cory/tmp/23-almond/extendedTrackList.json does not exist
2019-05-18 16:34:09,974 [http-nio-8080-exec-4] WARN  apollo.TrackService  - File /scratch_cory/tmp/23-almond/extendedTrackList.json can not be read
2019-05-18 16:34:09,974 [http-nio-8080-exec-4] WARN  apollo.TrackService  - File /scratch_cory/tmp/23-almond/extendedTrackList.json can not be written to
```

**Comment by alpapan** [comment]
See #2157 for access

**Comment by alpapan** [comment]
Ok, got it to work without the indexed fasta (just fasta).
Not sure what happened there and why it was not working with the indexed fasta.

```
{
   "formatVersion" : 1,
   "dataset_id" : "almond_s3",
   "names" : {
      "type" : "Hash",
      "url" : "https://s3.hortgenomics.science/jbrowse/almond/data/names/"
   },
   "tracks" : [
      {
         "category" : "Reference sequence",
         "chunkSize" : 20000,
         "key" : "Reference sequence",
         "label" : "DNA",
         "seqType" : "dna",
         "storeClass" : "JBrowse/Store/Sequence/StaticChunked",
         "type" : "SequenceTrack",
         "UrlTemplate" : "https://s3.hortgenomics.science/jbrowse/almond/data/seq/{refseq_dirpath}/{refseq}-",
      }
   ]
}
```

**Comment by alpapan** [comment]
I take it back... it seems that S3 doesn't as it used to...

**Comment by alpapan** [comment]
The problem was the case of the urltemplate after I removed the index fasta file.

The older version - on docker - worked with the faiUrlTemplate approach. I couldn't get the git version to work with that.

Then I tried without the indexed fasta and left it as UrlTemplate instead of urlTemplate.... obviously the former didn't work. the latter did.

Having said all that, serving files from S3 is only allowed for tracks that are not used to create gene models due to  #2135. Note however, that it expects .txtz files even though "compress" : 0

I'm going to close this ticket as #2135 has all the info needed
